### PR TITLE
[cli] Add "Environment" column to `vc list`

### PR DESCRIPTION
**Pull request by cb1kenobi**
Also adds a new `--environment=[preview|production]` filter and ability to pipe deployment URLs to `stdout`!

image

```
vc list --environment preview
```

```
vc list --environment production
```

```
vc list --environment preview > preview_deployments.txt
```

### CODE REVIEW

File: packages/cli/src/commands/pull.ts [hunk 1]
```diff
@@ -35,6 +35,7 @@ const help = () => {
     'DIR'
   )}    Path to the global ${'`.vercel`'} directory
     -d, --debug                    Debug mode [off]
+    --git-branch                   Specify the Git branch to pull specific Environment Variables for
```
Comment (TooTallNate) [inline packages/cli/src/commands/pull.ts:38]: This seems unrelated to the PR?

Comment (cb1kenobi) [reply packages/cli/src/commands/pull.ts:38]: Um, what the

**Review by TooTallNate — APPROVED** [review]

File: packages/cli/src/commands/pull.ts (hunk 1 above)
Comment (cb1kenobi) [reply packages/cli/src/commands/pull.ts:38]: Yes, unrelated. I found that option was missing when I documented `--git-branch` last week. Want me to move this to a separate PR?

**Review by trek — COMMENTED** [review]
Tested locally and all looks good.
